app/javascript/controllers/banner_controller.js

```diff
@@ -1,20 +1,54 @@
-import { Controller } from "@hotwired/stimulus"
+import { Controller } from "@hotwired/stimulus";
 
-// Connects to data-controller="banner"
 export default class extends Controller {
+  static targets = ["banner", "dismissButton", "showButton"];
+
   connect() {
-    this.element.dataset.stimulus = "banner-connected";
+    this.dismissButtonTarget.addEventListener("click", this.dismiss.bind(this));
+    this.showButtonTarget.addEventListener("click", this.show.bind(this));
+
+    if (this.isBannerDismissed()) {
+      this.hideBanner();
+      this.showShowButton(true); // Show the chevron
+    } else {
+      this.showBanner();
+      this.hideShowButton();
+    }
+  }
+
+  dismiss() {
+    const version = this.dismissButtonTarget.dataset.version;
+    document.cookie = `dismissed_banner_version=${version}; path=/; max-age=31536000`; // 1 year
+    this.hideBanner();
+    this.showShowButton();
+  }
+
+  show() {
+    document.cookie = `dismissed_banner_version=; path=/; max-age=0`; // Clear the cookie
+    this.showBanner();
+    this.hideShowButton();
   }
 
-  setCookie({ params: { time } }) {
-    const date = new Date(),
-      expiresInDays = 30,
-      cookie_name = "hideBanner2=" + time + "; ";
+  isBannerDismissed() {
+    const version = this.dismissButtonTarget.dataset.version;
+    return document.cookie
+      .split("; ")
+      .some((cookie) => cookie === `dismissed_banner_version=${version}`);
+  }
+
+  hideBanner() {
+    this.bannerTarget.style.display = "none";
+  }
 
-    date.setTime(date.getTime() + (expiresInDays * 24 * 60 * 60 * 1000));
-    const expiresText = "expires=" + date.toUTCString() + "; ";
+  showBanner() {
+    this.bannerTarget.style.display = "block";
+  }
+
+  hideShowButton() {
+    document.getElementById("show-banner-container").style.display = "none";
```

Comment:
I have discovered it's unreliable to use `document.getElementById` in stimulus, i strongly recommend changing this.

The way to go is to make the container a Stimulus target, and just call `this.containerTarget` to access the container.

```suggestion
    this.containerTarget.classList.remove('d-block');
    this.containerTarget.classList.add('d-none');
```